server/controllers/userAuth.js

```diff
@@ -411,6 +416,8 @@ const changePassword = (req, res) => {
 
         return doc.save();
       }
+
+      throw new Error();
```

Comment:
Please send validation exception in this case - consider inverting if so the check if for `!validHash` and throws an error from inside.